### ISSUE TITLE
Numbered citations please! :-)

### ISSUE THREAD

**Issue by rowanc1** [executablebooks/mystjs#162]
From @drscotthawley:

> Seeing all the author names & years becomes a cumbersome eyesore when you're citing a lot of different papers in the course of a (introductory) paragraph. More compact numerical citation markers are strongly preferred for some of our use cases. :-)

I think that this would be a "site" template configuration (although I suppose it could be project/page level). Currently this is dealt with totally in the renderer/template.

Actions:
- [ ] Decide on where in the config this should go
- [ ] Look to other examples for naming of this configuration option so that we are close/in-family to existing tools.

@drscotthawley if you have experience with this in other tools, I would love some help in the research side. From an implementation perspective this should be pretty easy -- just want to think through what the config options are called, and if we want any other citation like config while we are at it.

**Comment by drscotthawley** [comment]
@rowanc1  Agreed! A "site-wide" style would suit my needs as well, e.g. something in the YAML front matter. I notice that when I export to latex & PDF using the arxiv-twocolumn template, I _do_ get numbered citations.  But I would also like them for the website version. So whether that's done via some `{cite:n}` or a global options, I don't care. :-)

**Comment by eurunuela** [comment]
It would be great if the front matter had options to for instance order references alphabetically.

**Comment by alcrene** [comment]
I happened upon this thread by accident today and thought I could add my 2 cents based on what I’ve seen in research practice:

- The citation style is always preserved throughout a document, therefore it makes sense to have it as a global option.
- The citation style is almost always prescribed by the journal, therefore *semantically* it makes the most sense to set this in the templates.
  - *Pragmatically*, people are not just writing full-fledged papers targeted at journals with packaged templates, but also class notes, reports, etc. Most often the workflow will be to pick a close-enough template, and tweak it as necessary. For such cases, I think it makes sense to allow setting the citation style to be overridden by a document option, e.g. in the header.
- The MyST roles `{cite:p}` and `{cite:t}` are consistent with the equivalent roles/commands in *sphinx-cite* and *natbib*. Especially *natbib* AFAIK is still considered good (best?) practice and is used in official journal templates (such as those in the Physical Review family). Consistency with it is a Good Thing™, both because it makes MyST easier to learn / port to, but also because we know that the feature set of *natbib* is very mature.
  I think therefore that if new citation roles are to be introduced, they should line up with equivalent *natbib* commands. And if *natbib* does not contain equivalent commands, that should be considered strong indications that the role is probably not needed and may be detrimental.
